test/compileFunction.test.js

```diff
@@ -0,0 +1,271 @@
+    {name: 'pars', type: 'number', extends: true, description: 'Description of parameters'}
+  ]
+  t.deepEqual(actual, expected, 'extensible parameter')
+
+  t.end()
+})
+
+testAsync('compile function: @return', async t => {
+  let code, expected, actual
+
+  code = `function func (){}`
+  actual = _getReturn(compileJavascript(code))
+  expected = undefined
+  t.deepEqual(actual, expected, 'method.return comes only with doc')
+
+  code = `
+  /**
+   * @return {typeReturn} Description of return
+   */
+  function func (a, b){}
+  `
+  actual = _getReturn(compileJavascript(code))
+  expected = {type: 'typeReturn', description: 'Description of return'}
+  t.deepEqual(actual, expected, '@return with type')
+
+  t.end()
+})
+
+testAsync('compile function: @example', async t => {
+  let code, expected, actual
+
+  code = `
+  /**
+   * @example func(ex1)
+   * @example <caption>Example 2 function</caption> func(ex2)
+   */
+  function func (a, b){}
+  `
+  actual = _getExamples(compileJavascript(code))
+  expected = [
+    {
+      usage: 'func(ex1)'
+    }, {
+      usage: 'func(ex2)',
+      caption: 'Example 2 function'
+    }
+  ]
+  t.deepEqual(actual, expected, 'method.return comes only with doc')
+
+  t.end()
+})
+
+testAsync('compile function: kitchensink', async t => {
+  let code, expected, actual
+
+  code = `
+  /**
+   * Function description
+   *
+   * @title Function title
+   * @summary Function summary
+   *
+   * @example <caption>Example caption</caption>
+   *
+   * funcname(1, 2, 3, 4)
+   *
+   * @example
+   *
+   * funcname(x, y, z)
+   *
+   * @param  {par1Type} par1 Parameter one description
+   * @param  {...any} par2 Parameter two description
+   * @return {returnType} Return description
+   */
+  function funcname(par1, ...par2){
+    return par1 + sum(par2)
+  }
+  `
+  actual = _getFunc(compileJavascript(code))
+  expected = {
+    type: 'function',
+    code,
+    name: 'funcname',
+    title: 'Function title',
+    summary: 'Function summary',
+    description: 'Function description',
+    methods: {
+      'funcname(par1: par1Type, par2: any): returnType': {
+        signature: 'funcname(par1: par1Type, par2: any): returnType',
+        params: [
+          {
+            name: 'par1',
+            type: 'par1Type',
+            description: 'Parameter one description'
+          }, {
+            name: 'par2',
+            repeats: true,
+            type: 'any',
+            description: 'Parameter two description'
+          }
+        ],
+        return: {
+          type: 'returnType',
+          description: 'Return description'
+        },
+        examples: [
+          {
+            usage: 'funcname(1, 2, 3, 4)',
+            caption: 'Example caption'
+          }, {
+            usage: 'funcname(x, y, z)'
+          }
+        ]
+      }
+    }
+  }
+  t.deepEqual(actual, expected, 'method.return comes only with doc')
+
+  t.end()
+})
+
+// TODO: While we want this feature on the long term it seems that
```

Comment:
I think we should retain the ability to be able to document multiple methods for a function. As we've discussed before, it's important for documenting functions like `plot` and `add` which have several alternative parameter signatures. The current implementation works with adjoining doc comments like in this test example (which my research suggested seem to be the defacto standard way of doing overloaded functions with jsdoc). So, I'd prefer not to loose this feature right now. An alternative could be to merge methods coming from jsdoc strings having a common @name`